students/brandon_nguyen/session03/mailroom.py

```diff
@@ -0,0 +1,114 @@
+import sys
+import unittest
+from decimal import Decimal
+
+#!/usr/bin/env python3
+#Week3 Excercise mailroom part 1
+#Student: Brandon Nguyen - Au2018
+
+#global define data structure
+donor_db = [("Brandon Nguyen", [4500.33,350.87,300.05]),
+          ("Shawn DeGraw", [5500.00, 345.05,571.33]),
+          ("Jacqueline Lee",[4300.00,3200.00,230.13]),
+          ("Aidan Nguyen", [4300.00,200.00,238.23]),
+          ]
+promptTxt = "\n".join(("Welcome to the mailroom program!",
+          "Please choose from below 3 options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Quit",
+          ">>> "))
+#The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+
+#think about a main menu function that can be called any where.
+def main_menu():
+    promptTxt = "\n".join(("Welcome to the mailroom program!",
+          "Please choose from below 3 options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Quit",
+          ">>> "))
+
+def send_ty():
+    while True:
+        response = input("\nPlease enter donor full names or \"list\". Q to return to main menu: " )
+        if response.strip() == 'list':
+            list_donors()
+            break
+        elif response.strip() == 'Q':
+            main()
+        else:
+            for name in donor_db:
+            #print(name[0])
+                if name[0] == response.strip():
+                    indx = donor_db.index(name)
+                    while True:
+                        donation = input("Please enter donation amount for " + name[0] +". Q to quit. :>>")
+                        if donation.strip() == 'Q':
+                            main()
+                        else:
+                            break
+                    donor_db[indx][1].append(float(donation))
+                    print("\n\nHello {},\nThank you so much for the generious donation of {}!\n\n".format(name[0],float(donation)))
+                    main() #go to main menu not ideal way.  Should have another function that does the menu option
+            else:
+                donor_db.append((response.strip(),[0.00])) #adding new name with $0 donation. avoiding div/0 issue
+      
+def create_rpt():
+    """
+    This function is to print the report of donor.  
+    """
+    #Currently need more work on the format as 0.0 problem not yet solved.
+    print()
+    newList = sort_sum() #creating a new list with sorted total in reverse already for easy printing.
+    header_string = "Donor Name              |  Total Given | Num Gifts | Average Gift"
+    line = '-'*len(header_string)
+    print(header_string)
+    print(line)
+    for donor, total, num, avg in newList:
+        print( donor, "\t\t$ ",str(total[0]).rjust(13), str(num[0]).rjust(5), "\t   $ ", str(avg[0]).rjust(10) )
```

Comment:
It's recommended not to use `\t`, and this formatting better be achieved using various `"...{...}...{...}...".format(...)` approach (or Python 3.6+'s new f-strings as well).